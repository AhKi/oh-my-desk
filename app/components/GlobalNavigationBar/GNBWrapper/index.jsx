import React from 'react';
import GlobalNavigationBar from '../';
import './GNBWrapper.scss';

function GNBWrapper(WrappedComponent) {
	return function WrappedFunction(props) {
		return (
			<div className="GNBWrapper">
				<GlobalNavigationBar {...props} />
				<WrappedComponent {...props} />
			</div>
		);
	};
}

export default GNBWrapper;