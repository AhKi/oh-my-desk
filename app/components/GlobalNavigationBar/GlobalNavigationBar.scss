.GlobalNavigationBar {
  background-color: #2F3136;
  height: 100%;
  float: left;
  padding: 20px 0 0 20px;
  width: 30%;
  max-width: 218px;
  min-width: 125px;
  position: relative;
  margin: 0;
  display: block;
}

.GlobalNavigationBar__list {
  color: #BBBDC0;
  background-color: #2F3136;
  border: 0;
  height: 100%;
  text-align: center;
  width: 100%;
}

.GlobalNavigationBar__link {
  width: 100%;
  height: 100%;
  text-decoration-line: none;
}