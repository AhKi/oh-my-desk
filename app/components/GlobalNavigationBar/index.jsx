import React from 'react';
import PropTypes from 'prop-types'; // eslint-disable-line
import { Link } from 'react-router-dom';

const propTypes = {

};

class GlobalNavigationBar extends React.Component {
	render() {
		return (
			<div>
				<ul>
					<li>
						<Link to="/Setting">전체 설정</Link>
					</li>
					<li>
						<Link to="/widget-list">위젯</Link>
					</li>
					<li>
						<Link to="/widget-store">스토어</Link>
					</li>
				</ul>
			</div>
		);
	}
}

GlobalNavigationBar.propTypes = propTypes;

export default GlobalNavigationBar;