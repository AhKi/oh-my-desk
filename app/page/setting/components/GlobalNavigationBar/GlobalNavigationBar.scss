$GNB-background-color: #424242;

.GlobalNavigationBar {
  // flex-item
  order: 1;
  flex: 1 1 70px;
  background-color: $color-dark-gray-blue;
  max-width: 200px;
  transition: all .4s ease-in-out;
}

.GlobalNavigationBar__title {
  color: white;
  text-align: center;
  padding: 15px 0;
  border-bottom: 1px solid $color-cool-gray900;
}

.GlobalNavigationBar__list {
  color: #BBBDC0;
  height: 55px;
  text-align: center;
  width: 100%;
  border-bottom: 1px solid $color-cool-gray900;
  box-shadow: 0 1px 0 0 rgba(50, 56, 66,100);
}

.GlobalNavigationBar__list-active {
  background-color: $color-gray-blue;
  .GlobalNavigationBar__list-btn {
      color: $color-white;
    }

  path {
    fill: $color-white;
  }
}

.GlobalNavigationBar__list-btn {
  width: 100%;
  height: 100%;
  color: $color-cool-gray500;
  display: flex;
  align-items: center;
  font-size: 15px;
  font-weight: 400;
}

.GlobalNavigationBar__list-icon {
  display: inline-block;
  flex-grow: 1;
}

.GlobalNavigationBar__list-text {
  flex-basis: 52px;
  text-align: left;
}

.GlobalNavigationBar__list-arrow {
  flex-grow: 2;
  display: flex;
  justify-content: flex-end;
  text-align: right;

  animation-name: listArrow;
  animation-timing-function: cubic-bezier(0.65,-0.19, 0.34, 1.13);
  animation-fill-mode: both;
  animation-duration: 3s;
}

@keyframes listArrow {
  0% {
    opacity: 0;
    transform: translateX(-10px);
  }

  20% {
    opacity: 1;
    transform: translateX(0);
  }

  100% {
    transform: translateX(0);
  }
}

@media (max-width: 700px) {
  .GlobalNavigationBar {
    max-width: 60px;
  }

  .GlobalNavigationBar__logo {
    width: 30px;
  }

  .GlobalNavigationBar__list-text,
  .GlobalNavigationBar__list-arrow {
    display: none;
  }
}