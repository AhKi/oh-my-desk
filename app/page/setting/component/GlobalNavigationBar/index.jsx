import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import cx from 'classnames';
import './GlobalNavigationBar.scss';

const propTypes = {
  match: PropTypes.shape({
    path: PropTypes.string,
  }),
  onListClick: PropTypes.func,
  onSettingClick: PropTypes.func,
  onStoreClick: PropTypes.func,
};

const defaultProps = {
  match: {},
  onListClick() {},
  onSettingClick() {},
  onStoreClick() {},
};

class GlobalNavigationBar extends React.Component {
  static listClassName(current, target) {
    let isActive;

    if (typeof target === 'string') {
      isActive = current === target;
    } else if (Array.isArray(target)) {
      isActive = target.indexOf(current) !== -1;
    } else {
      isActive = false;
    }

    return cx(
      'GlobalNavigationBar__list',
      {
        'GlobalNavigationBar__list-active': isActive,
      },
    );
  }

  render() {
    const menu = [
      {
        icon: '',
        path: '/widget-list',
        name: 'Widget',
        match: ['/widget-list', '/widget-setting'],
        onClick: this.props.onListClick,
      },
      {
        icon: '',
        path: '/setting',
        name: 'Setting',
        onClick: this.props.onSettingClick,
      },
      {
        icon: '',
        path: '/widget-store',
        name: 'Widget store',
        onClick: this.props.onStoreClick,
      },
    ];

    return (
      <div className="GlobalNavigationBar">
        <div className="GlobalNavigationBar__title">
          OhMyDesk
        </div>
        <ul className="GlobalNavigationBar__menu-box">
          {menu.map(v => (
            <Link to={v.path} key={v.path}>
              <li className={GlobalNavigationBar.listClassName(
                this.props.match.path,
                v.match || v.path,
              )}
              >
                <button
                  className="GlobalNavigationBar__list-btn"
                  type="button"
                  onClick={v.onClick}
                >
                  {v.name}
                </button>
              </li>
            </Link>
          ))}
        </ul>
      </div>
    );
  }
}

GlobalNavigationBar.propTypes = propTypes;
GlobalNavigationBar.defaultProps = defaultProps;

export default GlobalNavigationBar;
