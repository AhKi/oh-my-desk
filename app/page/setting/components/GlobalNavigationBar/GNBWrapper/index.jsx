import React from 'react';
import GlobalNavigationBar from '..';
import './GNBWrapper.scss';

function GNBWrapper(WrappedComponent) {
  return function WrappedFunction(props) {
    return (
      <div className="GNBWrapper">
        <GlobalNavigationBar {...props} />
        <div className="GNBWrapper__content">
          <WrappedComponent {...props} />
        </div>
      </div>
    );
  };
}

export default GNBWrapper;
