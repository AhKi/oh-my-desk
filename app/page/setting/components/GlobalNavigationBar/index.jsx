import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import cx from 'classnames';
import logo from 'assets/logo/logo-white.svg';
import WidgetIcon from 'assets/icon/icon-menu-widget';
import StoreIcon from 'assets/icon/icon-menu-store';
import SettingIcon from 'assets/icon/icon-menu-setting';
import ArrowIcon from 'assets/icon/icon-menu-arrow';
import './GlobalNavigationBar.scss';

const propTypes = {
  match: PropTypes.shape({
    path: PropTypes.string,
  }),
  onListClick: PropTypes.func,
  onSettingClick: PropTypes.func,
  onStoreClick: PropTypes.func,
};

const defaultProps = {
  match: {},
  onListClick() {},
  onSettingClick() {},
  onStoreClick() {},
};

class GlobalNavigationBar extends React.Component {
  static calculateIsActive(current, target) {
    let isActive;

    if (typeof target === 'string') {
      isActive = current === target;
    } else if (Array.isArray(target)) {
      isActive = target.indexOf(current) !== -1;
    } else {
      isActive = false;
    }

    return isActive;
  }

  static listClassName(current, target) {
    let isActive;

    if (typeof target === 'string') {
      isActive = current === target;
    } else if (Array.isArray(target)) {
      isActive = target.indexOf(current) !== -1;
    } else {
      isActive = false;
    }

    return cx(
      'GlobalNavigationBar__list',
      {
        'GlobalNavigationBar__list-active': isActive,
      },
    );
  }

  render() {
    const { match } = this.props;
    const iconClassName = (current, target) => cx('GlobalNavigationBar__list-icon', {
      'GlobalNavigationBar__list-icon--active': GlobalNavigationBar.calculateIsActive(current, target),
    });
    const arrowClassName = (current, target) => cx('GlobalNavigationBar__list-arrow', {
      'GlobalNavigationBar__list-arrow--active': GlobalNavigationBar.calculateIsActive(current, target),
    });
    const menu = [
      {
        icon: WidgetIcon,
        path: '/widget-list',
        name: <span className="GlobalNavigationBar__list-text">Widget</span>,
        match: ['/widget-list', '/widget-setting'],
        onClick: this.props.onListClick,
      },
      {
        icon: SettingIcon,
        path: '/setting',
        name: <span className="GlobalNavigationBar__list-text">Setting</span>,
        onClick: this.props.onSettingClick,
      },
      {
        icon: StoreIcon,
        path: '/widget-store',
        name: <span className="GlobalNavigationBar__list-text">Store</span>,
        onClick: this.props.onStoreClick,
      },
    ];

    return (
      <div className="GlobalNavigationBar">
        <div className="GlobalNavigationBar__title">
          <img src={logo} alt="white-logo" className="GlobalNavigationBar__logo" />
        </div>
        <ul className="GlobalNavigationBar__menu-box">
          {menu.map((v) => {
            const Icon = v.icon;

            return (
              <Link to={v.path} key={v.path}>
                <li className={GlobalNavigationBar.listClassName(
                  this.props.match.path,
                  v.match || v.path,
                )}
                >
                  <button
                    className="GlobalNavigationBar__list-btn"
                    type="button"
                    onClick={v.onClick}
                  >
                    <Icon
                      className={iconClassName(match.path, v.match || v.path)}
                    />
                    <img
                      src={v.icon}
                      alt=""
                    />
                    {v.name}
                    <ArrowIcon
                      className={arrowClassName(match.path, v.match || v.path)}
                    />
                  </button>
                </li>
              </Link>
            );
          })}
        </ul>
      </div>
    );
  }
}

GlobalNavigationBar.propTypes = propTypes;
GlobalNavigationBar.defaultProps = defaultProps;

export default GlobalNavigationBar;
