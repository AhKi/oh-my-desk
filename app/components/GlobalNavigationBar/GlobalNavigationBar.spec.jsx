import React from 'react';
import { shallow } from 'enzyme';

import GlobalNavigationBar from './';

describe('<GlobalNavigationBar />', () => {
  it('should match to snapshot when render default', () => {
    const wrapper = shallow(<GlobalNavigationBar />);

    expect(wrapper).toMatchSnapshot();
  });
});
