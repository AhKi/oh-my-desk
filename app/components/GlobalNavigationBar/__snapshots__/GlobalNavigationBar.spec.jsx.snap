// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GlobalNavigationBar /> should match to snapshot when render default 1`] = `
<div
  className="GlobalNavigationBar"
>
  <ul
    className="list-group"
  >
    <Link
      className="GlobalNavigationBar__link"
      key="/setting"
      replace={false}
      to="/setting"
    >
      <li
        className="list-group-item list-group-item-action GlobalNavigationBar__list"
      >
        Setting
      </li>
    </Link>
    <Link
      className="GlobalNavigationBar__link"
      key="/widget-list"
      replace={false}
      to="/widget-list"
    >
      <li
        className="list-group-item list-group-item-action GlobalNavigationBar__list"
      >
        Widget
      </li>
    </Link>
    <Link
      className="GlobalNavigationBar__link"
      key="/widget-store"
      replace={false}
      to="/widget-store"
    >
      <li
        className="list-group-item list-group-item-action GlobalNavigationBar__list"
      >
        Widget store
      </li>
    </Link>
  </ul>
</div>
`;
